compiler/rustc_errors/src/emitter.rs

```diff
@@ -2175,30 +2175,69 @@ impl EmitterWriter {
         file_lines: &FileLines,
         is_multiline: bool,
     ) {
-        // Print the span column to avoid confusion
-        buffer.puts(*row_num, 0, &self.maybe_anonymized(line_start + line_pos), Style::LineNumber);
         if let DisplaySuggestion::Diff = show_code_change {
             // Add the line number for both addition and removal to drive the point home.
             //
             // N - fn foo<A: T>(bar: A) {
             // N + fn foo(bar: impl T) {
+            let number_of_lines = file_lines.lines.len();
+            for (index, line_to_remove) in
+                file_lines.lines.iter().take(number_of_lines - 1).enumerate()
+            {
+                buffer.puts(
+                    *row_num - 1,
+                    0,
+                    &self.maybe_anonymized(line_start + line_pos + index),
+                    Style::LineNumber,
+                );
+                buffer.puts(*row_num - 1, max_line_num_len + 1, "- ", Style::Removal);
+                buffer.puts(
+                    *row_num - 1,
+                    max_line_num_len + 3,
+                    &normalize_whitespace(
+                        &file_lines.file.get_line(line_to_remove.line_index).unwrap(),
+                    ),
+                    Style::NoStyle,
+                );
+                *row_num += 1;
+            }
+            let last_line = &file_lines
+                .file
+                .get_line(file_lines.lines[number_of_lines - 1].line_index)
+                .unwrap();
+            if last_line != line {
```

Comment:
Explain why we're doing this, something along the lines of "So far, we displayed the removal of one or more lines of code in the suggestion. If it wasn't a single line, then it means there's a partial removal or change in this last line, so we show the diff of that line as well."